src/Microsoft.AspNet.Mvc.ViewFeatures/Rendering/Html/DefaultEditorTemplates.cs

```diff
@@ -288,18 +281,26 @@ public static IHtmlContent ObjectTemplate(IHtmlHelper htmlHelper)
                     readOnly: false,
                     additionalViewData: null);
 
-                content.Append(templateBuilder.Build());
-
                 if (!propertyMetadata.HideSurroundingHtml)
                 {
-                    content.Append(" ");
-                    content.Append(htmlHelper.ValidationMessage(
+                    var divTag = new TagBuilder("div");
+                    divTag.AddCssClass("editor-field");
+
+                    var innerContent = new BufferedHtmlContent();
```

Comment:
Not sure what refactoring to avoid `TagRenderModel.StartTag` and `EndTag` buys us. This extra buffer and the new lack of alignment between the code and generated HTML are downsides.